File: go/05_security_misconfiguration.go
// A05 - Security Misconfiguration. VULN: pprof on public port, permissive TLS.
package main

import (
	_ "net/http/pprof"
	"net/http"
)

func misconfig() {
	// VULN: A05 - exposes pprof on 0.0.0.0
	go http.ListenAndServe(":6060", nil)
}

File: c/05_security_misconfiguration.c
/* A05 - Security Misconfiguration. VULN: debug symbols, permissive umask. */
#include <sys/stat.h>

void init_server(void) {
  umask(0); /* VULN: A05 world-writable files */
#ifdef DEBUG
  setenv("LD_LIBRARY_PATH", ".", 1); /* VULN: hijack */
#endif
}
